app/static/src/app/store/run/mutations.ts

```diff
@@ -122,5 +129,36 @@ export const mutations: MutationTree<RunState> = {
         if (paramSet) {
             paramSet.hidden = !paramSet.hidden;
         }
+    },
+
+    [RunMutation.SaveParameterDisplayName](state: RunState, payload: ParameterSetNames) {
```

Comment:
We don't usually have so much validation code in mutations, which are generally pretty dumb. I think we sometimes put it in actions if they have anything else to do, and I think in HINT validation often tends to live in utils and be invoked from the component (which feels like the wrong responsibility, but does have the benefit of not needing to clutter the store with ephemeral error messages). 

Not sure what the canonically correct location for Vue(x) validation code is - if you search for it you just get loads of suggestions for npm validation packages... 

I think it probably is right for this to be in the mutation now.  I guess we could have a more specialised validation module later if validation gets too verbose to live in mutations.

Reply:
hmm okay, i always viewed actions as async so more towards API request end and bits of synchronous code can just be in mutations but that is probably not right :laughing: so wait shall we leave it here for now then?